### ISSUE TITLE
global listening doesn't seem to work on Ventura

### ISSUE THREAD

**Issue by Pomax** [flit/MidiKeys#37]
I'm running Midi Keys with a global enable/disable hotkey, but when I have global input enabled, any typing I do still gets sent into whatever application I'm running instead of being intercepted and turned into MIDI events.

**Comment by Pomax** [comment]
Okay so that's not true... the listener for the global hotkey _does_ work, but whether it's on or off, typing anything without midikeys in focus does not get those keys routed through midikeys.

**Comment by xoxo369** [comment]
they dont care about us... let me know if you find a better alternative..

**Comment by Pomax** [comment]
Please remember that unpaid open source developers are humans too: if they no longer have time to look at this repo, then they don't have time to look at this repo. Unless you paid for the product, which you haven't, blaming them for not responding to an issue is grossly misunderstanding the relationship between you as a user and them as maintainer. 

If they ever find the time to look at this: great. If not: this issue will stay here for others to find (and maybe even motivate them to fix it themselves, and file a PR, or fork the project).

**Comment by xoxo369** [comment]
> Please remember that unpaid open source developers are humans too: if they no longer have time to look at this repo, then they don't have time to look at this repo. Unless you paid for the product, which you haven't, blaming them for not responding to an issue is grossly misunderstanding the relationship between you as a user and them as maintainer.
> 
> If they ever find the time to look at this: great. If not: this issue will stay here for others to find (and maybe even motive them to fix it themselves, and file a PR, or fork the project).

yeah I offered them money they don't care, we need to find another way to get it updated

**Comment by Pomax** [comment]
So pay someone else to fork it and update it. It's open source.

**Comment by xoxo369** [comment]
> So pay someone else to fork it and update it. It's open source.

how do I go about that?

**Comment by xoxo369** [comment]
still upset about this to be honest

**Comment by elektronen** [comment]
Hey folks. I understand there's a small group of people here that love this app, and have to come to rely on it. I am one of those people as well. The developer has every right to move on to other interests. Nobody does independent coding for free to be shackled to their creation for the rest of their lives.

What I'd like to do is encourage everyone to be constructive, not endlessly complain and berate the developer. Trying to guilt the dev into continuing work on a past passion project is a losing proposition.

Working with git is fairly easy. Forking the code is also easy. Figuring out how Apple changed the Accessibility UI portion of macOS between Big Sur and Ventura seems to be the thing to focus on here. AFAICS, MidiKeys always leveraged the Accessibility UI in order to provide the Global Hot Keys functionality.

**Comment by xoxo369** [comment]
Do you think apple blocked that kind of scripts because it could be used as
a keylogger? Where can i find a developer to pay to fork it and fix it?
Also listen to my music guys! @sleepindesigner on SoundCloud

**Comment by elektronen** [comment]
No, security in an operating system is rarely that targeted. The most
common cause of issues for older applications is going to be something
Apple changed to broadly improve security or, more likely, a portion of the
Accessibility UI that was rewritten to support input device portability.
The ability to move keyboard and mouse between computers and mobile devices
was introduced right after Big Sur.

There's probably a change log available for the version of XCode released
with macOS 12 or 13, and that would be a good place to start.

**Comment by xoxo369** [comment]
So with that being said how should I go about finding the right developers
to fork this and update it? I am genuinely willing to pay up to 100$ for it
to work again how it used to..

**Comment by elektronen** [comment]
What you're describing is called a code bounty. While $100 is no small
amount, it's one hour of time for an experienced developer, and two hours
of time for an honest carpenter. You can do some research and see what you
find.

**Comment by six-two** [comment]
I had the same issue and tried to look at the code. Sadly I have no macOS specific programming knowledge, so I did not find the relevant snippet. What I noticed is that the global shortcuts where you also need to hold modifiers (like `ctrl+alt+m` instead of `m`) do work for me. You can enable them here:

<img width="400" height="500" alt="Image" src="https://github.com/user-attachments/assets/17054101-13e2-4bb6-9f5b-3cb073e3fb54" />

So I made a Karabiner Elements config that will map keys like `m` into `ctrl+alt+m`. With this workaround it works pretty much as expected. If you would like to try it out, here is my config file. You can paste it in an advanced modification in Karabiner Elements: <https://gist.github.com/six-two/5442c8e3ff6a0e4287a04910a63a46fb>

You can toggle the rule on and off by pressing `ctrl+escape`. By default it is disabled. So you would press `ctrl+escape` to enable it, then press your keys `m`, `n`, `b`, ... and finally disable it with `ctrl+escape` when you are done.

If I missed a key or you want different modifiers, you can play around with the script that I built to generate the above config file. It is hosted here: <https://gitlab.com/six-two/karabiner-config/-/blob/main/rules/midikeys.py>
